#VRML_SIM R2023b utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/floors/protos/RectangleArena.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/appearances/protos/BrushedAluminium.proto"

WorldInfo {
  info [
    "Example use of a color Camera device."
  ]
  title "Camera"
}
Viewpoint {
  orientation -0.027358108409965148 -0.9812863595172179 -0.19060066770521528 5.5782644432094095
  position -2.1002348864511875 -0.1262060235045007 1.4810016485910633
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  floorTileSize 2 2
  wallAppearance BrushedAluminium {
    colorOverride 0.180392 0.180392 0.180392
  }
}
PointLight {
  attenuation 0 0 1
  intensity 0.6
  location 0 0 0.6
}
Robot {
  translation 0.09308833002239736 -0.3887622259466284 -0.0002268300820218866
  rotation -0.014409565100880031 -0.0006743301442201183 0.999895949442971 -3.0481787501129665
  children [
    TouchSensor {
      translation 0.045045045443021924 1.3910287068319427e-06 0.019984049189444526
      rotation 0.9999999999999999 -2.132423431373176e-08 -1.0188712379519262e-08 -3.141592290552515
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 0
            roughness 0.6980390000000001
            metalness 0
          }
          geometry DEF BUMPER Box {
            size 0.01 0.05 0.02
          }
        }
      ]
      name "bumper"
      boundingObject USE BUMPER
      physics Physics {
        density -1
        mass 0.01
      }
      lookupTable [
        0 0 0
        0.01 1 0
      ]
      linearVelocity -0.02489060202637722 -0.00233174973303838 -1.692684842279395e-09
      angularVelocity 4.433292745350954e-06 3.4837327522514947e-07 -3.3305332145669845e-06
    }
    Pose {
      translation 0 0 0.0415
      rotation 0 0 1 -1.570796
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.0820075 0.364731 0.8
            roughness 1
            metalness 0
          }
          geometry DEF BODY Cylinder {
            height 0.08
            radius 0.045
          }
        }
      ]
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 455.3009279107803
        axis 0 1 0
        anchor 0 0 0.025
      }
      device [
        RotationalMotor {
          name "left wheel motor"
          consumptionFactor 70
        }
        PositionSensor {
          name "left wheel sensor"
        }
      ]
      endPoint Solid {
        translation 0 0.04499999999999549 0.025
        rotation -0.695750244480732 0.6957504321175034 -0.17850191459283835 2.7883096327266426
        children [
          DEF WHEEL Pose {
            rotation 0.577656895078166 0.5771968951617173 -0.5771968951617173 -2.093935
            children [
              Shape {
                appearance PBRAppearance {
                  baseColor 1 0 0
                  roughness 1
                  metalness 0
                }
                geometry Cylinder {
                  height 0.01
                  radius 0.025
                }
              }
            ]
          }
        ]
        name "left wheel"
        boundingObject USE WHEEL
        physics DEF PHYSICS_WHEEL Physics {
          density -1
          mass 0.05
        }
        linearVelocity -0.024890822536799882 -0.0023319499799871993 -1.9478376303290968e-07
        angularVelocity 0.09329224254926939 -0.9956388265999226 -4.583406158327548e-06
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 2479.8174977266985
        axis 0 1 0
        anchor 0 0 0.025
      }
      device [
        RotationalMotor {
          name "right wheel motor"
          consumptionFactor 70
        }
        PositionSensor {
          name "right wheel sensor"
        }
      ]
      endPoint Solid {
        translation 0 -0.044999999999994156 0.025
        rotation -0.6331280498103233 0.6331279911549215 0.4453064330991754 3.9794804019330567
        children [
          USE WHEEL
        ]
        name "right wheel"
        boundingObject USE WHEEL
        physics USE PHYSICS_WHEEL
        linearVelocity -0.024890349128451717 -0.002331903098706099 1.973813574260523e-07
        angularVelocity 0.09329231532908257 -0.9956387961614923 -4.9231360730255146e-06
      }
    }
    DistanceSensor {
      translation 0.042 0.02 0.063
      rotation 0 0 -1 -0.499205
      children [
        DEF SENSOR Pose {
          rotation 0.577656895078166 0.5771968951617173 -0.5771968951617173 -2.093935
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.975691 0.981481 0.0252992
                roughness 1
                metalness 0
              }
              geometry Cylinder {
                height 0.004
                radius 0.008
              }
            }
          ]
        }
      ]
      name "ds0"
      lookupTable [
        0 1024 0
        0.05 1024 0
        0.15 0 0
      ]
      numberOfRays 2
      aperture 1
    }
    DistanceSensor {
      translation 0.042 -0.02 0.063
      rotation 0 0 1 -0.500795
      children [
        USE SENSOR
      ]
      name "ds1"
      lookupTable [
        0 1024 0
        0.05 1024 0
        0.15 0 0
      ]
      numberOfRays 2
      aperture 1
    }
    DEF SMILE Pose {
      translation 0.036 0 0.025
      rotation 0.25056300542381216 -0.9351130202419243 -0.25056300542381216 1.637834
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.721569 0.290196 0.290196
            roughness 1
            metalness 0
          }
          geometry Cylinder {
            height 0.018
            radius 0.009
            subdivision 3
          }
        }
      ]
    }
    Camera {
      translation 0.04 0 0.0915
      children [
        Pose {
          translation -0.015 0 0
          rotation -0.5771968951617173 -0.577656895078166 -0.5771968951617173 -2.093935
          children [
            Shape {
              appearance PBRAppearance {
                roughness 1
                metalness 0
              }
              geometry Cylinder {
                height 0.03
                radius 0.01
              }
            }
          ]
        }
      ]
      fieldOfView 1.047
      width 80
      height 60
      antiAliasing TRUE
      noise 0.04
    }
  ]
  name "MyBot"
  boundingObject Pose {
    translation 0 0 0.0415
    rotation 0 0 1 -1.570796
    children [
      USE BODY
    ]
  }
  physics Physics {
    density -1
    mass 0.5
  }
  controller "HW3_Q3_FSA"
  linearVelocity -0.024890580280296025 -0.0023319980851635998 1.2760651633350114e-09
  angularVelocity 4.433292745251781e-06 3.483732747884965e-07 -3.3305332146186476e-06
}
Solid {
  translation 0.283461 0.28857 0.07
  rotation 0 0 1 -0.261795307179586
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 1 0 0
        metalness 0
      }
      geometry Cylinder {
        height 0.15
        radius 0.035
      }
    }
  ]
  name "solid(1)"
  boundingObject Cylinder {
    height 0.15
    radius 0.035
  }
}
Solid {
  translation -0.130033 -0.432109 0.07
  rotation 0 0 1 -0.261795307179586
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 1 1 0
        metalness 0
      }
      geometry Cylinder {
        height 0.15
        radius 0.035
      }
    }
  ]
  name "solid(2)"
  boundingObject Cylinder {
    height 0.15
    radius 0.052
  }
}
Solid {
  translation -0.448282 0.449922 0.035
  rotation 0 0 1 4.692820414042842e-06
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0 0 1
        metalness 0
      }
      geometry Box {
        size 0.1 0.1 0.07
      }
    }
  ]
  boundingObject Box {
    size 0.1 0.1 0.07
  }
}
Solid {
  translation -0.260165 0.0912033 0.07
  rotation 0 0 1 -0.261795307179586
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0 1 0
        metalness 0
      }
      geometry Cylinder {
        height 0.15
        radius 0.035
      }
    }
  ]
  name "solid(3)"
  boundingObject Cylinder {
    height 0.15
    radius 0.035
  }
}
